utils/staking-miner/src/main.rs

```diff
@@ -190,15 +210,42 @@ impl MinerProfile {
 	}
 }
 
-#[derive(Debug, Clone)]
+#[derive(Debug, thiserror::Error)]
 enum Error {
-	KeyFileNotFound,
-	KeyFileCorrupt,
+	Io(#[from] std::io::Error),
+	Jsonrpsee(#[from] jsonrpsee_ws_client::Error),
+	Codec(#[from] codec::Error),
+	Crypto(sp_core::crypto::SecretStringError),
+	RemoteExternalities(&'static str),
+	PalletMiner(EPM::unsigned::MinerError),
+	PalletElection(EPM::ElectionError),
```

Comment:
```suggestion
	Crypto(#[from] sp_core::crypto::SecretStringError),
	RemoteExternalities(&'static str),
	PalletMiner(#[from] EPM::unsigned::MinerError),
	PalletElection(#[from] EPM::ElectionError),
```

You don't need the manual implementations for these errors.

Reply:
I have to try again, I think I ran into an issue.

Reply:
ah yeah so the inner errors must implement `StdError` apparently?

Reply:
Hmm, I thought that requirement triggered on `#[derive(thiserror::Error)]`, not on the `#[from]`. But yeah, error-handling is a lot better IMO when all of them implement `std::error::Error`.

Reply:
I'll resolve this for now. Moving all our errors to implement `std::error:Error` might be a good substrate vision issue. Although, I am not sure if this is in `sp_std`, and if not, with the removal of the native runtime it will become kinda pointless.